src/main/java/programminglife/model/drawing/SubGraph.java

```diff
@@ -285,35 +285,26 @@ public boolean checkDynamicLoad(int leftBorder, double rightBorder) {
      * @return int of the new centerNode.
      */
     public int updateCenterNode(double centerCanvasX, int oldCenterNode) {
-        for (int i = 1; i < layers.size(); i++) {
-            if (layers.get(i).getX() > centerCanvasX) {
-                if (Math.abs(layers.get(i - 1).getX() - centerCanvasX)
-                        < Math.abs(layers.get(i).getX() - centerCanvasX)) {
-                    for (DrawableNode node : layers.get(i - 1)) {
-                        if (node instanceof DrawableSegment
-                                && ((DrawableSegment) node).getSequence().length() > 1) {
-                            return node.getIdentifier();
-                        }
-                    }
-                    for (DrawableNode node : layers.get(i)) {
-                        if (node instanceof  DrawableSegment
-                                && ((DrawableSegment) node).getSequence().length() > 1) {
-                            return node.getIdentifier();
-                        }
-                    }
-                } else {
-                    for (DrawableNode node : layers.get(i)) {
-                        if (node instanceof DrawableSegment
-                                && ((DrawableSegment) node).getSequence().length() > 1) {
-                            return node.getIdentifier();
-                        }
-                    }
-                    for (DrawableNode node : layers.get(i - 1)) {
-                        if (node instanceof  DrawableSegment
-                                && ((DrawableSegment) node).getSequence().length() > 1) {
-                            return node.getIdentifier();
-                        }
-                    }
+        int centerIndex = getLayerIndex(layers, centerCanvasX);
+        Layer centerLayer = layers.get(centerIndex);
+        for (DrawableNode node : centerLayer) {
+            if (node instanceof DrawableSegment
+                    && ((DrawableSegment) node).getSequence().length() > 1) {
+                return node.getIdentifier();
+            }
+        }
+        if (centerIndex >= layers.size()) {
```

Comment:
Too little, too late: line 289 would have thrown an IndexOutOfBoundsException before this check.
Also, getLayerIndex is guaranteed to return an index that is actually within bounds of the list.